src/main/java/seedu/edulink/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,62 @@
+package seedu.edulink.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.edulink.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.function.Predicate;
+
+import seedu.edulink.commons.util.ToStringBuilder;
+import seedu.edulink.logic.Messages;
+import seedu.edulink.model.Model;
+import seedu.edulink.model.student.Student;
+
+/**
+ * Filters list of student based on tags.
+ * Can have multiple tags involved.
+ */
+public class FilterCommand extends Command {
+
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose tag matches exaclty with "
+            + "the specified tags and displays them as a list with index numbers.\n"
+            + "Parameters: " + PREFIX_TAG + "KEYWORD "
+            + "[" + PREFIX_TAG + "KEYWORD] ... \n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_TAG + "TA" + " " + PREFIX_TAG + "Kind";
+
+    private final Predicate<Student> predicate;
+
+    public FilterCommand(Predicate<Student> predicate) {
+        this.predicate = predicate;
+    }
```

Comment:
Should we put all the logic in FilerCommandParser.java? Would it be better to put some logic in FilterCommand.java?

Reply:
This is done to follow the implementation of the FindCommand which was an existent class beforehand. I think it is best to follow the same implementation as to not confuse future developers.